build.sbt

```diff
@@ -21,6 +21,8 @@ sonatypeProfileName := "org.virtuslab"
 
 Global / onChangedBuildSource := ReloadOnSourceChanges
 
+(Compile / compile) := (Compile / compile).dependsOn(checkCurrentJavaVersion).value
```

Comment:
Actually.... would it make sense to put the check into `initialize ~= { _ => ... }` block instead? 🤔

Reply:
I replaced this Compile / compile line with initialize block - please take a look